### ISSUE TITLE
Proton does not work with i3wm

### ISSUE THREAD

**Issue by zbirenbaum** [ValveSoftware/Proton#6300]
I use i3wm and a number of other utilities to supply the features of a desktop environment I desire, but after tons of time debugging the fact that literally no version of proton other than 4.11 would ever work, despite all my drivers being fine, and wine always working fine, I decided to try installing a more conventional desktop environment (xfce4 with lxdm) to see if that helped any.

It completely fixed the problems with steam, but the issue is that I don't want all the junk that desktop environments install on my system, nor do I want to have to completely reconfigure my system just so that proton works.... Is there any known workaround for this? If not, is there some reliance on a session manager or something that could be initialized along with the window manager?

**Comment by kisak-valve** [comment]
Hello @zbirenbaum, this issue should also be mentioned to the window manager's dev(s).

**Comment by MayeulC** [comment]
Hello @zbirenbaum FWIW, it works (and has been working for years) with sway. I use sway on Arch (steam native, runtime or flatpak do not affect this). I don't even use a login manager, starting sway from a tty every time. The only notable adjustment I made is that I `export $(dbus-launch)` before lanching sway.

I've seen reports of people using i3wm around here, so that's definitely not the missing piece. Your tone in the second paragraph comes across as a bit rude though: you aren't the only one trying to keep your system lean; if steam is missing a dependency it would be an issue with your distribution packages. If it's a runtime dependency, it should be documented.

Could you please post your steam log? That issue might be due to something else. Did leaving xfce4 installed make it work on i3.
How did you install steam? What distribution? Could you add your system info (from Steam->Help->System Info) as an attachment? Thank you.
IIRC, proton > 4.11 starts to make use of namespace-based sandboxing, which can cause issues with flatpak (in this case, use community builds) or older kernels.

**Comment by zbirenbaum** [comment]
> Hello @zbirenbaum, this issue should also be mentioned to the window manager's dev(s).

I tested out other WMs and seem to have the same issue, so I am not sure it is an i3 problem. I can make an issue post there in case anyone else has run into this though. Other people using it don't seem to have a problem according to @MayeulC, so maybe the issue needs to be re-titled, though I'm not what to re-title it too. 

> Hello @zbirenbaum FWIW, it works (and has been working for years) with sway. I use sway on Arch (steam native, runtime or flatpak do not affect this). I don't even use a login manager, starting sway from a tty every time. The only notable adjustment I made is that I `export $(dbus-launch)` before lanching sway.

I tried out sway today since I have seen a couple of reports that it works well under it. While I can't say for sure that it wasn't working properly, the behavior for the other games I had installed that didn't work didn't change at all. A couple were non-steam installation games though, so it is very possible that proton didn't apply some game specific fixes that might have been otherwise.

I launch i3 with `exec dbus-launch i3`, which I believe should accomplish the same thing as exporting `$(dbus-launch)`. 

> I've seen reports of people using i3wm around here, so that's definitely not the missing piece. Your tone in the second paragraph comes across as a bit rude though: you aren't the only one trying to keep your system lean; if steam is missing a dependency it would be an issue with your distribution packages. If it's a runtime dependency, it should be documented.

I apologize if it came across as rude, I didn't intend it that way. The last time I installed a DE was KDE around a year and a half ago, and I was still getting random hidden files generated in my home dir which were associated with KDE apps after uninstalling all of said apps and KDE itself. My problem with desktop environments isn't so much how much they install, it's how hard they are to properly uninstall. Hours of googling and trying to fix the issue couldn't resolve it, and I ended up having to back up my files and do a clean install for that system to stop having random KDE files for uninstalled applications appearing, so the subject is a bit touchy for me.

> Could you please post your steam log? That issue might be due to something else. Did leaving xfce4 installed make it work on i3. How did you install steam? What distribution? Could you add your system info (from Steam->Help->System Info) as an attachment? Thank you. IIRC, proton > 4.11 starts to make use of namespace-based sandboxing, which can cause issues with flatpak (in this case, use community builds) or older kernels.

I kept track of everything I installed including dependencies by copying pacman's output before installing xfce and lxdm. After noticing proton suddenly worked for Persona 5 Royal after starting it in xfce I then uninstalled everything using the list of packages I kept in a text file. The game still works perfectly now, so I don't think it is a dependency issue unless there is some initialization step that occurs the first time a game is run other than generating the prefix.

My distribution is Arch Linux, and I have had this issue across two laptops and a desktop all running the same distro. I installed steam through the Arch Linux multilib repository, and tried running it both through the `steam` and `steam-runtime` executables. I was told the flatpak might resolve my problem so I then installed that, but the behavior was the same.

The proton log last generated can be found here: https://pastebin.com/rEv3y56J
The system info generated by steam is here: https://pastebin.com/XRBgj8AP

In case it is helpful, here is the list of packages I installed (and later uninstalled). I'd like to note that just installing them wasn't enough, I also had to actually start the game in xfce for it to start working under i3. After that, the packages were seemingly irrelevant.
```
colord gnome-themes-extra libcanberra libdbusmenu-glib libdbusmenu-gtk3  libgtop libimobiledevice libplist libusbmuxd libwnck3 libxfce4ui libxfce4util libxklavier libxpresent libxres polkit-gnome sound-theme-freedesktop tdb upower usbmuxd xfce4-notifyd xorg-iceauth exo garcon thunar thunar-volman tumbler xfce4-appfinder xfce4-panel xfce4-power-manager xfce4-session xfce4-settings xfce4-terminal xfconf xfdesktop xfwm4 xfwm4-themes lxdm
```

**Comment by lennie420** [comment]
If not using a display manager then you would need to launch i3 with xorg-xinit. Configure a .xinitrc file and then launch using startx.

**Comment by zbirenbaum** [comment]
> If not using a display manager then you would need to launch i3 with xorg-xinit. Configure a .xinitrc file and then launch using startx.

Yes, that is how I do it.

**Comment by zbirenbaum** [comment]
As an update, I believe I discovered the problem. There were a few of them so it took a bit to iron out, but my experience now matches Protondb generally.

The primary issue was that the modesetting driver wasn't being utilized for some reason, despite being the one defined in my xorg config. Anyone else with this issue should entirely uninstall the xf86-video-intel driver on top of making sure that modesetting is the driver being loaded by xorg. 

Second, there are some periodic crashes and general instabilities while using the nvidia card for iGPU render offloading, but I've looked into it and it is an issue with the Intel and nvidia drivers themselves. Gotta love proprietary code...

Third, there are some instances where passing VK_ICD_FILENAMES rather than leaving it undefined fixed a crash on start.

Finally, as a semi related tip, if you are swapping to the modesetting driver and plan to use nvidia for rendering despite the crashing issues, make sure you enable compositing in Xorg config and use your compositor to enable vsync, otherwise you will have the worst screen tearing you'll ever experience in every application.

Since I resolved my own issue I'll go ahead and close this.
